server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/KeywordsConsistencyOnDelegationMailbox.feature

```diff
@@ -0,0 +1,189 @@
+
+  Scenario: Get message list should keep flags on non-shared mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags $Draft via messageIdProbe of "mess" on "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailbox "notShared" with flag "$Draft"
+    Then the message list is empty
+
+  Scenario: Get message list should keep flags on shared mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags $Draft via messageIdProbe of "mess" on "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailbox "shared" with flag "$Draft"
+    Then the message list has size 1
+    And the message list contains "mess"
+
+  Scenario: getMessage with shared user should return message with combine flag when delegation mailbox
+    Given "alice@domain.tld" is connected
+    And the user set flags $Flagged via messageIdProbe of "mess" on "shared" of user "alice@domain.tld"
+    When "bob@domain.tld" is connected
+    And the user ask for messages "mess"
+    Then no error is returned
+    And the keywords of the message is $Flagged
+
+  Scenario: getMessage of owner mailbox should return message with combine flag when delegation mailbox
+    Given "alice@domain.tld" is connected
+    And the user set flags $Flagged via messageIdProbe of "mess" on "shared" of user "alice@domain.tld"
+    When the user ask for messages "mess"
+    Then no error is returned
+    And the keywords of the message is $Flagged
+
+  Scenario: getMessage should keep origin message status when cut the sharing
+    Given "alice@domain.tld" is connected
+    And the user set flags $Flagged via messageIdProbe of "mess" on "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" does not share its mailbox "shared" with "bob@domain.tld"
+    And "bob@domain.tld" is connected
+    And the user ask for messages "mess"
+    Then no error is returned
+    And the empty keywords of the message
+
+  Scenario: message should update message status based on delegation mailbox
+    Given "alice@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged,$Seen"
+    And "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Seen"
+    When "alice@domain.tld" is connected
+    And the user ask for messages "mess"
+    Then no error is returned
+    And the message is "\Flagged \Seen" on "notShared"
+    And the message is "\Seen" on "shared"
+
+  Scenario: message should keep origin message status when cut the sharing
+    And "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged"
+    When "alice@domain.tld" does not share its mailbox "shared" with "bob@domain.tld"
+    And "alice@domain.tld" is connected
+    And the user ask for messages "mess"
+    Then no error is returned
+    And the message is "\Flagged" on "shared"
+    And the message is "" on "notShared"
```

Comment:
A stepdefs named "And the message has no flags on "noShared"" would have been more understandable